src/dashboard/script.ts

```diff
@@ -0,0 +1,84 @@
+/// <reference lib="dom" />
+
+import { FetchPageTest } from "./api-test";
+
+async function loadUsers(pageCursor: number) {
+  const page = await FetchPageTest(pageCursor);
+  if (!page) return;
+
+  for (const user of page) {
+    const podiumHTML = `<img src="${user.avatar}"><span onclick="navigator.clipboard.writeText('${user.userid}')">${user.username}</span>`;
+    const xpInfoHTML = `<span class="xp-popup">${user.progress[0]}/${Math.floor((user.progress[0] * 100) / user.progress[1])}</span>`;
+    if (user.pos === 1) {
+      const first = document.querySelector<HTMLDivElement>("#first")!;
+      first.querySelector("h2")!.innerHTML = podiumHTML;
+      first.querySelector("#total-xp-level")!.innerHTML =
+        `Level: ${user.level}<br>Total XP: ${user.xp}`;
+      first.querySelector("#xp")!.innerHTML = xpInfoHTML;
+      continue;
+    }
+
+    if (user.pos === 2) {
+      const second = document.querySelector<HTMLDivElement>("#second")!;
+      second.querySelector("h2")!.innerHTML = podiumHTML;
+      second.querySelector("#total-xp-level")!.innerHTML =
+        `Level: ${user.level}<br>Total XP: ${user.xp}`;
+      second.querySelector("#xp")!.innerHTML = xpInfoHTML;
+      continue;
+    }
+
+    if (user.pos === 3) {
+      const third = document.querySelector<HTMLDivElement>("#third")!;
+      third.querySelector("h2")!.innerHTML = podiumHTML;
+      third.querySelector("#total-xp-level")!.innerHTML =
+        `Level: ${user.level}<br>Total XP: ${user.xp}`;
+      third.querySelector("#xp")!.innerHTML = xpInfoHTML;
+      continue;
+    }
+
+    const listings = document.querySelector("section")!;
+    listings.insertAdjacentHTML(
+      "beforeend",
+      `<div class="user">
+      <p class="pos">${user.pos}</p>
+      <p class="name" onclick="navigator.clipboard.writeText('${user.userid}')"><img src="${user.avatar}"><span>${user.username}</span></p>
+      <label>
+        <p><span>Level: ${user.level}</span><span>XP: ${user.xp}</span></p>
+        <progress value="${user.progress[0]}" max="${(user.progress[0] * 100) / user.progress[1]}">${Math.floor((user.progress[0] * 100) / user.progress[1])}</progress>
+        ${xpInfoHTML}
+      </label>
+    </div>`,
+    );
+  }
+}
+
+let pageCursor = 0;
+const maxPage = 10;
```

Comment:
A maxPage will not work in production code. However, after removing it, the user should get some sort of error message when they scroll all the way down, for example "Sorry lad, you reached the end!"

Reply:
I was gonna replace that with the `GetMaxPage()` function you told me to use.